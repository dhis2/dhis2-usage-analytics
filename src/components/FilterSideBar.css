.uaa-filter-sidebar {
    flex: 0 0 280px;
    background-color: #ffffff;
    padding: 1rem;
    border-right: 1px solid rgba(0, 0, 0, 0.12);
}
.uaa-filter-sidebar .uaa-app-header {
    font-size: 2rem;
    margin-bottom: 2rem;
}
