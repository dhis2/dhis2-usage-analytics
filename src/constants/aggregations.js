import i18n from '@dhis2/d2-i18n'

export const SUM = 'SUM'
export const AVERAGE = 'AVERAGE'

const AGGREGATIONS = {
    [SUM]: { value: SUM, label: i18n.t('Sum') },
    [AVERAGE]: { value: AVERAGE, label: i18n.t('Average') },
}
export default AGGREGATIONS
