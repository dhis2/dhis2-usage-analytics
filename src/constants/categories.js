import i18n from '@dhis2/d2-i18n'

export const FAVORITE_VIEWS = 'FAVORITE_VIEWS'
export const FAVORITES_SAVED = 'FAVORITES_SAVED'
export const USERS = 'USERS'
export const TOP_FAVORITES = 'TOP_FAVORITES'
export const DATA_VALUES = 'DATA_VALUES'

const CATEGORIES = [
    { value: FAVORITE_VIEWS, label: i18n.t('Favorite views') },
    { value: FAVORITES_SAVED, label: i18n.t('Favorites saved') },
    { value: USERS, label: i18n.t('Users') },
    { value: TOP_FAVORITES, label: i18n.t('Top favorites') },
    { value: DATA_VALUES, label: i18n.t('Data values') },
]
export default CATEGORIES
