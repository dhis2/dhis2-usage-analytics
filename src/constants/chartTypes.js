import i18n from '@dhis2/d2-i18n'

export const ALL = 'ALL'
export const TOTAL = 'TOTAL'

const chartTypes = [
    { value: ALL, label: i18n.t('All') },
    { value: TOTAL, label: i18n.t('Total') },
]

export default chartTypes
