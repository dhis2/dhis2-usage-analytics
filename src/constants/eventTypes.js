import i18n from '@dhis2/d2-i18n'

export const CHART_VIEW = 'CHART_VIEW'
export const MAP_VIEW = 'MAP_VIEW'
export const REPORT_TABLE_VIEW = 'REPORT_TABLE_VIEW'
export const EVENT_REPORT_VIEW = 'EVENT_REPORT_VIEW'
export const EVENT_CHART_VIEW = 'EVENT_CHART_VIEW'
export const DASHBOARD_VIEW = 'DASHBOARD_VIEW'
export const DATA_SET_REPORT_VIEW = 'DATA_SET_REPORT_VIEW'

const EVENT_TYPES = {
    [CHART_VIEW]: { value: CHART_VIEW, label: i18n.t('Chart') },
    [MAP_VIEW]: { value: MAP_VIEW, label: i18n.t('Map') },
    [REPORT_TABLE_VIEW]: {
        value: REPORT_TABLE_VIEW,
        label: i18n.t('Report table'),
    },
    [EVENT_REPORT_VIEW]: {
        value: EVENT_REPORT_VIEW,
        label: i18n.t('Event report'),
    },
    [EVENT_CHART_VIEW]: {
        value: EVENT_CHART_VIEW,
        label: i18n.t('Event chart'),
    },
    [DASHBOARD_VIEW]: { value: DASHBOARD_VIEW, label: i18n.t('Dashboard') },
    [DATA_SET_REPORT_VIEW]: {
        value: DATA_SET_REPORT_VIEW,
        label: i18n.t('Data set report'),
    },
}

export default EVENT_TYPES
