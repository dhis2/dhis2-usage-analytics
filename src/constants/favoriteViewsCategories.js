import i18n from '@dhis2/d2-i18n'

export const SUM_ALL = 'SUM_ALL'
export const SUM_TOTAL = 'SUM_TOTAL'
export const AVEGAGE_ALL = 'AVEGAGE_ALL'
export const AVERAGE_TOTAL = 'AVERAGE_TOTAL'

const FAVORITE_VIEWS_CATEGORIES = [
    { id: SUM_ALL, displayName: i18n.t('All favorite views') },
    { id: SUM_TOTAL, displayName: i18n.t('Total views') },
    {
        id: AVEGAGE_ALL,
        displayName: i18n.t('All favorite average views per user'),
    },
    { id: AVERAGE_TOTAL, displayName: i18n.t('Total views per user') },
]
export default FAVORITE_VIEWS_CATEGORIES
