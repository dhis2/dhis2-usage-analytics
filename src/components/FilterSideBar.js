import React, { Fragment } from 'react'
import PropTypes from '@dhis2/prop-types'
import { connect } from 'react-redux'
import i18n from '@dhis2/d2-i18n'
import * as Fields from './FilterFields'
import { FAVORITE_VIEWS, TOP_FAVORITES } from '../constants/categories'
import './FilterSideBar.css'

export function FilterSideBar({
    showDateFields,
    showFavoriteViewsFields,
    showTopFavoritesFields,
}) {
    return (
        <aside className="uaa-filter-sidebar">
            <h1 className="uaa-app-header">{i18n.t('Usage Analytics')}</h1>
            <Fields.CategoryDropDown />
            {showDateFields && (
                <Fragment>
                    <Fields.DateRange />
                    <Fields.IntervalDropDown />
                </Fragment>
            )}
            {showFavoriteViewsFields && (
                <Fragment>
                    <Fields.AggregationLevelDropDown />
                    <Fields.ChartTypeDropDown />
                </Fragment>
            )}
            {showTopFavoritesFields && (
                <Fragment>
                    <Fields.EventTypeDropDown />
                    <Fields.PageSizeDropDown />
                    <Fields.SortOrderDropDown />
                </Fragment>
            )}
        </aside>
    )
}

FilterSideBar.propTypes = {
    showDateFields: PropTypes.bool,
    showFavoriteViewsFields: PropTypes.bool,
    showTopFavoritesFields: PropTypes.bool,
}

export function mapStateToProps({ filter: { category } }) {
    return {
        showDateFields: category.value !== TOP_FAVORITES,
        showTopFavoritesFields: category.value === TOP_FAVORITES,
        showFavoriteViewsFields: category.value === FAVORITE_VIEWS,
    }
}

export default connect(mapStateToProps)(FilterSideBar)
