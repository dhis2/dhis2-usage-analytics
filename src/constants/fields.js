import * as CATS from './categories'
import * as FAV_VIEW_CATS from './favoriteViewsCategories'
import * as PROPS from './properties'

const FAVORITE_VIEWS_TOTAL_FIELDS = [
    PROPS.date,
    PROPS.mapViews,
    PROPS.chartViews,
    PROPS.pivotTableViews,
    PROPS.eventReportViews,
    PROPS.eventChartViews,
    PROPS.dashboardViews,
    PROPS.dataSetReportViews,
    PROPS.totalViews,
]

const FAVORITE_VIEWS_AVERAGE_FIELDS = [
    PROPS.date,
    PROPS.averageMapViews,
    PROPS.averageChartViews,
    PROPS.averagePivotTableViews,
    PROPS.averageEventReportViews,
    PROPS.averageEventChartViews,
    PROPS.averageDashboardViews,
    PROPS.averageViews,
]

export const FIELDS = {
    [CATS.FAVORITE_VIEWS]: {
        [FAV_VIEW_CATS.SUM_ALL]: FAVORITE_VIEWS_TOTAL_FIELDS,
        [FAV_VIEW_CATS.SUM_TOTAL]: FAVORITE_VIEWS_TOTAL_FIELDS,
        [FAV_VIEW_CATS.AVERAGE_ALL]: FAVORITE_VIEWS_AVERAGE_FIELDS,
        [FAV_VIEW_CATS.AVERAGE_TOTAL]: FAVORITE_VIEWS_AVERAGE_FIELDS,
    },
    [CATS.FAVORITES_SAVED]: [
        PROPS.date,
        PROPS.savedMaps,
        PROPS.savedCharts,
        PROPS.savedPivotTables,
        PROPS.savedEventReports,
        PROPS.savedEventCharts,
        PROPS.savedDashboards,
        PROPS.savedIndicator,
    ],
    [CATS.USERS]: [PROPS.date, PROPS.activeUsers, PROPS.users],
    [CATS.TOP_FAVORITES]: [
        PROPS.position,
        PROPS.name,
        PROPS.views,
        PROPS.id,
        PROPS.created,
    ],
    [CATS.DATA_VALUES]: [PROPS.date, PROPS.savedDataValues],
}
export default FIELDS
