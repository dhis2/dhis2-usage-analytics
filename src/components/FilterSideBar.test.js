import React from 'react'
import { shallow } from 'enzyme'
import { FilterSideBar, mapStateToProps } from './FilterSideBar'
import { FAVORITE_VIEWS, TOP_FAVORITES, USERS } from '../constants/categories'

jest.mock('react-redux', () => ({
    connect: jest.fn(() => component => `Connected${component.name}`),
}))

const defaultState = { filter: { category: FAVORITE_VIEWS } }

describe('<FilterSideBar/>', () => {
    it('Matches the snapshot for default category: FAVORITE_VIEWS', () => {
        const props = mapStateToProps(defaultState)
        const tree = shallow(<FilterSideBar {...props} />)
        expect(tree).toMatchSnapshot()
    })
    it('Matches the snapshot for category: TOP_FAVORITES', () => {
        const state = { ...defaultState, filter: { category: TOP_FAVORITES } }
        const props = mapStateToProps(state)
        const tree = shallow(<FilterSideBar {...props} />)
        expect(tree).toMatchSnapshot()
    })
    it('Matches the snapshot for any other category, i.e. USERS', () => {
        const state = { ...defaultState, filter: { category: USERS } }
        const props = mapStateToProps(state)
        const tree = shallow(<FilterSideBar {...props} />)
        expect(tree).toMatchSnapshot()
    })
})
