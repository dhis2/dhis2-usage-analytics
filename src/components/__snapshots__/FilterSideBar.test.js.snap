// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSideBar/> Matches the snapshot for any other category, i.e. USERS 1`] = `
<aside
  className="uaa-filter-sidebar"
>
  <h1
    className="uaa-app-header"
  >
    Usage Analytics
  </h1>
  <Connect(CategoryDropDown) />
  <Connect(StartDateInput) />
  <Connect(StartDateInput) />
  <Connect(IntervalDropDown) />
</aside>
`;

exports[`<FilterSideBar/> Matches the snapshot for category: TOP_FAVORITES 1`] = `
<aside
  className="uaa-filter-sidebar"
>
  <h1
    className="uaa-app-header"
  >
    Usage Analytics
  </h1>
  <Connect(CategoryDropDown) />
  <Connect(EventTypeDropDown) />
  <Connect(PageSizeDropDown) />
  <Connect(SortOrderDropDown) />
</aside>
`;

exports[`<FilterSideBar/> Matches the snapshot for default category: FAVORITE_VIEWS 1`] = `
<aside
  className="uaa-filter-sidebar"
>
  <h1
    className="uaa-app-header"
  >
    Usage Analytics
  </h1>
  <Connect(CategoryDropDown) />
  <Connect(StartDateInput) />
  <Connect(StartDateInput) />
  <Connect(IntervalDropDown) />
  <Connect(AggregationLevelDropDown) />
  <Connect(ChartTypeDropDown) />
</aside>
`;
